lens/sqlrepo/blockstore.go

```diff
@@ -0,0 +1,235 @@
+package sqlrepo
+
+import (
+	"context"
+	"strings"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/lib/blockstore"
+	blocks "github.com/ipfs/go-block-format"
+	"github.com/ipfs/go-cid"
+	"github.com/multiformats/go-multibase"
+)
+
+var errNoRows = pgx.ErrNoRows
+
+var dbpool *pgxpool.Pool
+
+var log = logging.Logger("sql")
+
+func DB(connStr string) *pgxpool.Pool {
+	if dbpool == nil {
+		var err error
+		dbpool, err = pgxpool.Connect(context.Background(), connStr)
+		if err != nil {
+			log.Fatalf("failed to connect to %s: %s", connStr, err)
+		}
+
+		for _, ddl := range []string{
+			"CREATE TABLE IF NOT EXISTS blocks(" +
```

Comment:
Assuming this means blockstore blocks and not blockchain blocks -- would it be worth using the [Large Object type](https://www.postgresql.org/docs/10/lo-intro.html) here instead? This also may be the wrong place to ask this.

Reply:
cc @ribasushi

Reply:
really i suspect that this as a consumer shouldn't be making tables at all, but it's the same blockstore object that riba is using on the lotus producer side for writing into.

Reply:
@frrist it means blockstore-blocks-specific-to-the-blockchain: the largest block is something like 38k, definitely not a candidate for a large object

What @willscott is saying is correct re "this code should not be creating tables". Moreover: this is all highly WiP/MVP: the "schema" was for a test, it is **definitely** not how it actually should look...

If you want to incorporate this into the visor now ( the producer work is placed on hold for a few days ) - I recommend making the actual queries configurable. Then you do not care what the database looks like.

Reply:
Thanks for the solid explanation @ribasushi. I'm not meaning to be overly critical, just curious. Still happy to have this in as to avoid PR rot @willscott, my approval is sticky.

> it means blockstore-blocks-specific-to-the-blockchain: the largest block is something like 38k, definitely not a candidate for a large object.

Would the IPLD object representing the state tree at a specific state root also be in this store? Where is the code for this library?

Reply:
@frrist no worries, just giving context why the parts-to-be-critiqued exist at all. I wasn't expecting us to make this into an official visor plugin when I was playing with it ;)